rcl/test/rcl/test_timer.cpp

```diff
@@ -559,3 +568,113 @@ TEST_F(TestPreInitTimer, test_timer_get_allocator) {
+  EXPECT_EQ(times_called, 1);
+
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 3);
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_get_time_until_next_call(&timer, &next_call_end));
+  EXPECT_GT(next_call_end, next_call_start);
+
+  next_call_start = next_call_end;
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_exchange_period(&timer, 0, &old_period));
+  EXPECT_EQ(RCL_S_TO_NS(1), old_period);
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 4);
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_get_time_until_next_call(&timer, &next_call_end));
+  EXPECT_GT(next_call_start, next_call_end);
+
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_cancel(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(RCL_RET_TIMER_CANCELED, rcl_timer_call(&timer));
+  EXPECT_EQ(times_called, 4);
+}
+
+TEST_F(TestPreInitTimer, test_get_callback) {
+  ASSERT_EQ(timer_callback_test, rcl_timer_get_callback(&timer)) << rcl_get_error_string().str;
+}
+
+TEST_F(TestPreInitTimer, test_timer_reset) {
+  int64_t next_call_start, next_call_end;
+  times_called = 0;
+
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 2);
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_get_time_until_next_call(&timer, &next_call_start));
+
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_reset(&timer));
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_get_time_until_next_call(&timer, &next_call_end));
+  EXPECT_GT(next_call_start, next_call_end);
+
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_cancel(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(RCL_RET_TIMER_CANCELED, rcl_timer_call(&timer));
+  EXPECT_EQ(times_called, 2);
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_reset(&timer));
+  EXPECT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 3);
+}
+
+TEST_F(TestPreInitTimer, test_timer_exchange_callback) {
+  times_called = 0;
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 1);
+  ASSERT_EQ(
+    timer_callback_test, rcl_timer_exchange_callback(
+      &timer, timer_callback_changed)) << rcl_get_error_string().str;
+
+  ASSERT_EQ(RCL_RET_OK, rcl_timer_call(&timer)) << rcl_get_error_string().str;
+  EXPECT_EQ(times_called, 0);
+}
+
+TEST_F(TestPreInitTimer, test_invalid_get_guard) {
+  ASSERT_EQ(NULL, rcl_timer_get_guard_condition(nullptr));
+}
+
+TEST_F(TestPreInitTimer, test_invalid_init_fini) {
+  rcl_allocator_t bad_allocator = get_failing_allocator();
+  rcl_timer_t timer_fail;
+
+  EXPECT_EQ(
+    RCL_RET_ALREADY_INIT, rcl_timer_init(
+      &timer, &clock, this->context_ptr, 500, nullptr,
+      rcl_get_default_allocator())) << rcl_get_error_string().str;
+
+  timer_fail = rcl_get_zero_initialized_timer();
```

Comment:
@Blast545 consider collapsing declaration and definition into one line, here or above in line 649.